### PR TITLE
nodegl: rename class to cls

### PR DESCRIPTION
**Pull request by jmoguillansky-gpsw**
We should avoid using C++ keywords, especially in headers, so that
node.gl code can be used from both C and C++

### CODE REVIEW

**Comment by cboesch-gpsw** [comment]
We're already using `cls`; can we make that consistent?

**Comment by mbouron-gpsw** [comment]
@cboesch-gpsw  clazz is also used in some of our jni code.

**Comment by cboesch-gpsw** [comment]
> @cboesch-gpsw clazz is also used in some of our jni code.

Alright well then, let's decide on renaming either `cls` or `clazz` for consistency first.

**Comment by mbouron-gpsw** [comment]
I'm ok with `cls` if it's ok with everyone (and I'll make a commit to rename the jni part).

**Comment by jmoguillansky-gpsw** [comment]
sure I can change to cls

File: libnodegl/gctx.c [hunk 1]
```diff
@@ -27,12 +27,12 @@
 #include "nodes.h"
 #include "rendertarget.h"
 
-extern const struct gctx_class ngli_gctx_gl;
-extern const struct gctx_class ngli_gctx_gles;
+extern const struct gctx_cls ngli_gctx_gl;
+extern const struct gctx_cls ngli_gctx_gles;
```
Comment (cboesch-gpsw) [inline libnodegl/gctx.c:31]: I don't think you want to rename that struct

File: libnodegl/gctx.c [hunk 2]
```diff
@@ -60,51 +60,51 @@ struct gctx *ngli_gctx_create(const struct ngl_config *config)
             backend_map[config->backend].string_id);
         return NULL;
     }
-    const struct gctx_class *class = backend_map[config->backend].cls;
-    struct gctx *s = class->create(config);
+    const struct gctx_class *cls = backend_map[config->backend].cls;
```
Comment (cboesch-gpsw) [inline libnodegl/gctx.c:63]: especially since you're not renaming it here ^

File: libnodegl/pipeline.c [hunk 3]
```diff
@@ -25,57 +25,57 @@
...
 int ngli_pipeline_update_buffer(struct pipeline *s, int index, struct buffer *buffer)
 {
-    return s->gctx->class->pipeline_update_buffer(s, index, buffer);
+    return s->gctx->cls->pipeline_update_buffer(s, index, buffer);
 }
 
 void ngli_pipeline_draw(struct pipeline *s, int nb_vertices, int nb_instances)
 {
-    s->gctx->class->pipeline_draw(s, nb_vertices, nb_instances);
+    return s->gctx->cls->pipeline_draw(s, nb_vertices, nb_instances);
```
Comment (cboesch-gpsw) [inline libnodegl/pipeline.c:63]: that added return looks unrelated and wrong

File: libnodegl/pipeline.c [hunk 4]
```diff
@@ -25,57 +25,57 @@
...
 }
 
 void ngli_pipeline_draw_indexed(struct pipeline *s, struct buffer *indices, int indices_format, int nb_indices, int nb_instances)
 {
-    s->gctx->class->pipeline_draw_indexed(s, indices, indices_format, nb_indices, nb_instances);
+    return s->gctx->cls->pipeline_draw_indexed(s, indices, indices_format, nb_indices, nb_instances);
```
Comment (cboesch-gpsw) [inline libnodegl/pipeline.c:68]: idem added return

File: libnodegl/pipeline.c [hunk 5]
```diff
@@ -25,57 +25,57 @@
...
 }
 
 void ngli_pipeline_dispatch(struct pipeline *s, int nb_group_x, int nb_group_y, int nb_group_z)
 {
-    s->gctx->class->pipeline_dispatch(s, nb_group_x, nb_group_y, nb_group_z);
+    return s->gctx->cls->pipeline_dispatch(s, nb_group_x, nb_group_y, nb_group_z);
```
Comment (cboesch-gpsw) [inline libnodegl/pipeline.c:73]: idem added return

File: libnodegl/gctx.c (hunk 1 above)
Comment (jmoguillansky-gpsw) [reply libnodegl/gctx.c]: updated

File: libnodegl/gctx.c (hunk 2 above)
Comment (jmoguillansky-gpsw) [reply libnodegl/gctx.c:63]: updated

File: libnodegl/pipeline.c (hunk 3 above)
Comment (jmoguillansky-gpsw) [reply libnodegl/pipeline.c]: updated

File: libnodegl/pipeline.c (hunk 4 above)
Comment (jmoguillansky-gpsw) [reply libnodegl/pipeline.c]: updated

File: libnodegl/pipeline.c (hunk 5 above)
Comment (jmoguillansky-gpsw) [reply libnodegl/pipeline.c]: updated

**Review by mbouron-gpsw — APPROVED** [review]

**Comment by cboesch-gpsw** [comment]
Extract of `git grep '\<class\>'` in your branch:
```
libnodegl/gen_doc.c:#define CLASS_LIST(type_name, class) extern const struct node_class class;
libnodegl/gen_doc.c:#define TYPE2CLASS(type_name, class)            \
libnodegl/gen_doc.c:        extern const struct node_class class;   \
libnodegl/gen_doc.c:        return &class;                          \
libnodegl/gen_doc.c:static void print_node_type(struct bstr *b, const struct node_class *class)
libnodegl/gen_doc.c:    const char *class_ref = class->params_id ? class->params_id : class->name;
libnodegl/gen_doc.c:    ngli_bstr_printf(b, "[%s](#", class->name);
libnodegl/gen_doc.c:#define CLASS_COMMALIST(type_name, class) &class,
libnodegl/gen_specs.c:#define CLASS_LIST(type_name, class) extern const struct node_class class;
libnodegl/gen_specs.c:#define CLASS_COMMALIST(type_name, class) &class,
libnodegl/jni_utils.c:        LOG(ERROR, "could not find Throwable class");
libnodegl/jni_utils.c:        LOG(ERROR, "could not find Throwable class's class");
libnodegl/jni_utils.h: * Jni field describing a class, a field or a method to be retrieved using
libnodegl/jni_utils.h: * Retrieve class references, field ids and method ids to an arbitrary structure.
libnodegl/jni_utils.h: * the class/field/method to be retrieved
libnodegl/jni_utils.h: * Delete class references, field ids and method ids of an arbitrary structure.
libnodegl/jni_utils.h: * the class/field/method to be deleted
libnodegl/nodes.c:static struct ngl_node *node_create(const struct node_class *class)
libnodegl/nodes.c:    node = aligned_allocz(node_size + class->priv_size);
libnodegl/nodes.c:    node->cls = class;
libnodegl/nodes.c:#define REGISTER_NODE(type_name, class)         \
libnodegl/nodes.c:        extern const struct node_class class;   \
libnodegl/nodes.c:        ngli_assert(class.id == type_name);     \
libnodegl/nodes.c:        return &class;                          \
libnodegl/nodes.c:    const struct node_class *class = get_node_class(type);
libnodegl/nodes.c:    if (!class) {
libnodegl/nodes.c:    struct ngl_node *node = node_create(class);
```

**Review by mbouron-gpsw — CHANGES_REQUESTED** [review]
The name class is still used  is nodes.c gen_specs.c, gen_doc.c.

**Comment by mbouron-gpsw** [comment]
Also please reword the commit message to `internal: rename class to cls` and add a period at the end of the description.

**Comment by jmoguillansky-gpsw** [comment]
> Extract of `git grep '\<class\>'` in your branch:
> 
> ```
> libnodegl/gen_doc.c:#define CLASS_LIST(type_name, class) extern const struct node_class class;
> libnodegl/gen_doc.c:#define TYPE2CLASS(type_name, class)            \
> libnodegl/gen_doc.c:        extern const struct node_class class;   \
> libnodegl/gen_doc.c:        return &class;                          \
> libnodegl/gen_doc.c:static void print_node_type(struct bstr *b, const struct node_class *class)
> libnodegl/gen_doc.c:    const char *class_ref = class->params_id ? class->params_id : class->name;
> libnodegl/gen_doc.c:    ngli_bstr_printf(b, "[%s](#", class->name);
> libnodegl/gen_doc.c:#define CLASS_COMMALIST(type_name, class) &class,
> libnodegl/gen_specs.c:#define CLASS_LIST(type_name, class) extern const struct node_class class;
> libnodegl/gen_specs.c:#define CLASS_COMMALIST(type_name, class) &class,
> libnodegl/nodes.c:static struct ngl_node *node_create(const struct node_class *class)
> libnodegl/nodes.c:    node = aligned_allocz(node_size + class->priv_size);
> libnodegl/nodes.c:    node->cls = class;
> libnodegl/nodes.c:#define REGISTER_NODE(type_name, class)         \
> libnodegl/nodes.c:        extern const struct node_class class;   \
> libnodegl/nodes.c:        ngli_assert(class.id == type_name);     \
> libnodegl/nodes.c:        return &class;                          \
> libnodegl/nodes.c:    const struct node_class *class = get_node_class(type);
> libnodegl/nodes.c:    if (!class) {
> libnodegl/nodes.c:    struct ngl_node *node = node_create(class);
> ```

updated

**Comment by jmoguillansky-gpsw** [comment]
> Also please reword the commit message to `internal: rename class to cls` and add a period at the end of the description.

updated

**Comment by jmoguillansky-gpsw** [comment]
> The name class is still used is nodes.c gen_specs.c, gen_doc.c.

updated

**Comment by mbouron-gpsw** [comment]
LGTM, thanks.

**Review by mbouron-gpsw — APPROVED** [review]

**Pull request merged**